txscript/script.go

```diff
@@ -275,7 +275,9 @@ func parseScriptTemplate(script []byte, opcodes *[256]opcode) ([]parsedOpcode, e
 		retScript = append(retScript, pop)
 	}
 
-	return retScript, nil
+	ret := make([]parsedOpcode, len(retScript))
```

Comment:
This doesn't fall under the description of the PR.. I also don't really think it's necessary.

Reply:
This is a real P2SH transaction sample:
txid: 5da708e9a903ac5d41a0f3b0731bedf3a2f99522d9978ec4d7d93412157296ce
P2SH script:a914f323f8af174e6829a9c8acf1013f65f18370380987

I will output origin script length and final length of the  result slice in parseScriptTemplate() function. as following:

```
func parseScriptTemplate(script []byte, opcodes *[256]opcode) ([]parsedOpcode, error) {
fmt.Println("script length:", len(script))
         ....
fmt.Println("retScript length:", len(retScript))
return retScript, nil
```

And I using a unit testing here:
```
func TestRealScript(t *testing.T) {
	b, _ := hex.DecodeString("a914f323f8af174e6829a9c8acf1013f65f18370380987")
	_, err := parseScript(b)
	if err != nil {
		t.Error(err)
	}
}
```

And run the unit testing, output:
   script length: 23
   retScript length: 3

So I think there are the memory space of twenty structs(parsedOpcode) wasting!

If there is any error, please let me know.

Reply:
@stevenroose PTAL

Reply:
@qshuai I'm saying that the PR says "error annotation", while this addresses something else.

Looking at the entire method, it does seem to make sense to do the copy, but please don't mix several things in one PR.

Reply:
@stevenroose Yeah,  you are right. This PR include several commits. Github will merge all my commits in one PR. 
Is there a good practice? 

I am a newer bitcoin coder, and my english is poor. But I just want learn more. Please pardon my misnomer if has.🙂